cdhweb/people/exodus.py

```diff
@@ -98,21 +102,38 @@ def user_group_exodus():
     # had an unused "Content Editor" group that should be deleted
     Group.objects.exclude(pk__in=(moderators.pk, editors.pk)).delete()
 
-    # remove any users who had login disabled for some reason, except the
-    # script user
-    non_staff = 0
-    for user in User.objects.filter(Q(is_staff=False) | Q(is_active=False)) \
-                            .exclude(username=settings.SCRIPT_USERNAME):
-        non_staff += 1
-        logging.debug("removing user %s with is_staff=False" % user)
-        user.delete()
-    if non_staff:
-        logging.info("removed %d non-staff users" % non_staff)
-
-    # assign all remaining users to moderators group; remove superuser.
-    for user in User.objects.all():
+    # remove any users who have never logged in, except script user. this way
+    # old accounts can still be associated with their changes/log entries
+    no_actions = User.objects.filter(last_login__isnull=True) \
+                             .exclude(username=settings.SCRIPT_USERNAME)
+    logging.debug("removing %d users with no logins/changes" %
+                  no_actions.count())
+    no_actions.delete()
+
+    # for anyone without a current CDH position, mark user account as inactive.
+    # remove superuser permissions and add the user to the moderators group
+    for user in User.objects.exclude(username=settings.SCRIPT_USERNAME):
         logging.debug("adding user %s to wagtail moderators" % user)
         moderators.user_set.add(user)
         user.is_superuser = False
         user.save()
+        try:
+            person = Person.objects.get(user=user)
+            if not person.current_title:
+                logging.debug("marking user %s non-active" % user)
+                user.is_active = False
+                user.save()
```

Comment:
Seems like this might be possible to do in a queryset filter and update, but I guess it's valuable to see explicitly what's being done per user